presto-main/src/main/java/com/facebook/presto/sql/analyzer/Scope.java

```diff
@@ -58,13 +60,15 @@ private Scope(
             boolean queryBoundary,
             RelationId relationId,
             RelationType relation,
-            Map<String, WithQuery> namedQueries)
+            Map<String, WithQuery> namedQueries,
+            Map<Identifier, WindowSpecification> windowSpecifications)
     {
         this.parent = requireNonNull(parent, "parent is null");
         this.relationId = requireNonNull(relationId, "relationId is null");
         this.queryBoundary = queryBoundary;
         this.relation = requireNonNull(relation, "relation is null");
         this.namedQueries = ImmutableMap.copyOf(requireNonNull(namedQueries, "namedQueries is null"));
+        this.windowSpecifications = ImmutableMap.copyOf(requireNonNull(windowSpecifications, "windowSpecifications is null"));
```

Comment:
It may not be guaranteed by docs but it's how it's implemented.
I don't like the chain of `ImmutableMap.copyOf(requireNonNull...)` because it's quite long while typically it does not bring any value (unless Guava drops the requirement of non-nullness in some future version).
But I won't object if you insist of leaving it as is because I don't feel strongly about it.

Agreed, we have places in Presto codebase where we have `ImmutableX.copyOf(requireNonNull(...))` but we also used to use simply `ImmutableX.copyOf(...)` without an extra `requireNonNull` check inside.